SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriver.h

```diff
@@ -203,6 +188,8 @@ class SPATIALGDK_API USpatialNetDriver : public UIpNetDriver
 	TUniquePtr<SpatialGDK::WellKnownEntitySystem> WellKnownEntitySystem;
 	TUniquePtr<SpatialGDK::ClientConnectionManager> ClientConnectionManager;
 
+	TUniquePtr<FSpatialLoadBalancingHandler> LoadBalancingHandler;
```

Comment:
...and a TSharedPtr<FSpatialLoadBalancingHandler> here?

Reply:
If we make this a shared pointer, is that just to enable using weak pointer in the rep graph or is there more benefit? Even so, it does make sense, will do this

Reply:
It would enable actual shared ownership if we ever want - but so far it's just to enable TWeakPtr here https://github.com/spatialos/UnrealGDK/pull/2879#discussion_r572854668 so possible future crashes are less confusing :)

Reply:
makes sense- although personally I'm not sure shared ownership is a benefit here. Will add a comment explaining why this is a shared pointer.

Reply:
It seems to be an established pattern to use UniquePtr's for storing an object on the NetDriver, and raw pointers to keep a reference to the actual object; is there a reason behind this approach?

Reply:
I mean, I haven't seen a place where shared ownership would be required in our systems so far, but keeping a raw pointer to something on a completely different object, the lifetime of which is controlled by Unreal's GC, is scary nonetheless. 

I'm also pondering whether we can skip caching pointers to UniquePtr-owned systems on other objects and instead picked them up from the NetDriver every time we need them, as this would allow us to see that the NetDriver has expired.

Reply:
I agree that keeping raw pointers around on objects which is controlled by Unreal's GC is scary. That's why generally we are moving away from UObjects etc. so that we have a better understanding of the lifecycles of our classes. My mental model is lifecycles of most GDK classes that aren't the raw connection or Spatial View (maybe also the view even) should be tied to the SpatialNetDriver if possible, as GDK logic generally is related to the connection to a deployment. However, as you point out here, this makes sense as a weak pointer because the rep graph is a UClass. (At least, I think that's the reason it makes sense)

Skipping caching pointers- absolutely, great idea, some of the refactoring I've done has moved towards that approach. See the ActorSystem. Unfortunately we're still mid process, but the kind of setup you're describing sounds like the same end goal. 

Does this answer the question?

Reply:
Yeah this is awesome, thank you!